src/fhirtypes/Identifier.ts

```diff
@@ -0,0 +1,45 @@
+import { CodeableConcept } from './CodeableConcept';
+import { InvalidPeriodError } from '../errors/InvalidPeriodError';
+
+/**
+ * Represents the FHIR R4 data type Identifier.
+ * Instances of this type are used to uniquely identify
+ * some other entity within a system.
+ *
+ * @see {@link http://hl7.org/fhir/R4/datatypes.html#Identifier}
+ */
+export type Identifier = {
+  use?: string;
+  type?: CodeableConcept;
+  system?: string;
+  value?: string;
+  period?: Period;
+  assigner?: any;
+};
+
+/**
+ * Represents the FHIR R4 data type Period.
+ * This type includes a constraint on the start and end values,
+ * which are enforced in the following validatePeriod method.
+ *
+ * @see {@link http://hl7.org/fhir/R4/datatypes.html#Period}
+ */
+export type Period = {
+  start?: Date;
+  end?: Date;
+};
+
+/**
+ * Enforces the constraint on Period instances that the start
+ * cannot come after the end.
+ * @param period - the Period to validate.
+ * @returns {boolean} - true when the Period is valid.
+ * @throws {InvalidPeriodError} when the start comes before the end.
+ */
+
+export function validatePeriod(period: Period): boolean {
```

Comment:
I think it makes more sense to have these return `void` -- because as it is now, the only possible value it can ever return is `true`, so there's no reason for a caller to ever check the value.  By actually saying it returns a `boolean`, the caller may mistakenly assume that they should expect `true/false` to indicate if it validates (rather than a thrown exception).  If we return `void`, it's a bit more obvious that an error will throw (since the function would be pointless otherwise).